pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -92,9 +92,8 @@ def optimizer_step(
         if not isinstance(model, pl.LightningModule) or not model.automatic_optimization or not skipped_backward:
             optimizer.step(**kwargs)
 
-    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
```

Comment:
This will be a breaking change if a subclass is calling the super implementation, consider keeping an empty shell during deprecation

Reply:
I think it shouldn't be an empty-shell, but it should raise a warning and call either `state_dict` or `load_state_dict` for convenience. Usually that's what people want to do here :)

Reply:
good point, I think an empty shell will do bc lightning takes care of saving/loading states in `state_dict/load_state_dict` now.

The only need to keep the actual old `on_save/load_checkpoint` contents is if users manually check `ckpt["old_precplugin_key"]` somewhere in their code... do we need a deprecation process to support this? e.g. write precision plugin state_dict under both `"old_precplugin_key"` and under `qualname`